### DISCUSSION TITLE
GraphQL cursor based pagination - What needs to done?

### DISCUSSION THREAD

**Discussion by Harsgalt86** [ghiscoding/aurelia-slickgrid#1093] [Q&A]
As per title, what needs to be done to get GraphQL cursor based pagination fully implemented?

**Comment by ghiscoding** [comment]
I don't know, it's been way too long but from the comments I left in the GraphQL Service [here](https://github.com/ghiscoding/slickgrid-universal/blob/3a89dd2200ed38c7a05fe4e88ec06b6510679e11/packages/graphql/src/services/graphql.service.ts#L309-L328), it might already be in place and just missing the docs. If it doesn't work then I don't know and I have no use for it at the moment, so I'm not intending to go much further with this unfortunately.

**Reply by Harsgalt86** [reply]
I had a play around and from what I can tell it doesn't support the "after" (and "before" and "last") input when building the query.
I think it's a matter of updating it's internal state based on a returned `PageInfo` object. 

If I investigate and open a PR would you be open to reviewing and merging?

**Reply by ghiscoding** [reply]
Yes sure, I'm always open to PR as long as it doesn't lower test coverage (meaning adding or modifying unit tests is ideal if something changes). It does look like I already have some tests with cursor, so they might just need to be updated, for example here are some of these tests [here](https://github.com/ghiscoding/slickgrid-universal/blob/780753609fed398ffd706eb03115a640a69b1bdc/packages/graphql/src/services/__tests__/graphql.service.spec.ts#L200-L222)

**Reply by Harsgalt86** [reply]
Awesome thanks, have a PR open here - https://github.com/ghiscoding/slickgrid-universal/pull/1153
